### DISCUSSION TITLE
A Python playground using Brython

### DISCUSSION THREAD

**Discussion by hatemhosny** [brython-dev/brython#2222] [Show and tell]
Great project. Thank you!

Please allow me to introduce [LiveCodes](https://livecodes.io), a [feature-rich](https://livecodes.io/docs/features/), [open-source](https://github.com/live-codes/livecodes), [client-side](https://livecodes.io/docs/why/#client-side), code playground that supports [80+ languages and frameworks](https://livecodes.io/docs/why/#language-support). The [announcement post](https://blog.livecodes.io/introducing-livecodes/) gives a general overview.

Python is supported in LiveCodes using **Brython**.

A Python starter template is available for a quick start:  
https://livecodes.io/?template=python  
(see all [starter templates](https://livecodes.io/?new)).

You can start an empty Python project using this link: https://livecodes.io/?py

Projects can be [shared](https://livecodes.io/docs/features/share/), [exported](https://livecodes.io/docs/features/export/) and [deployed](https://livecodes.io/docs/features/deploy/) (to GitHub Pages).

Playgrounds can be [embedded](https://livecodes.io/docs/features/embeds) in any webpage, using a powerful, yet easy-to-use, [SDK](https://livecodes.io/docs/sdk).

This is an example for an embedded Python playground:

```html
<div id="container"></div>

<script type="module">
  import { createPlayground } from 'https://unpkg.com/livecodes';

  createPlayground('#container', {
    params: {
      python: 'from browser import document\ndocument["header"].html = f"Hello, Python!"\nprint("Good morning")',
      html: '<h1 id="header"></h1>',
      console: 'open',
    },
  });
</script>
```
[Preview](https://livecodes.io/?mode=result&x=code/N4IgLglmA2CmIC4QBkIDdYGED2ATWAzgAQDKAIgNJECiAHgIYC2ADnCADQj4EDGAThGaRsAO0QgO4egHMCiANoBdTvR6QM1XFGx9xjenwDWAV2aT9R04lDR6I6cZnwkACzCNoknqLCwRYcQAeLTQiCFwAXgAdEG9-eggRWD4YgD5AgHoQ1KiUkVyokUDeASEiMABPZlhokEY8Yzg0gpEiMJYdMCJgIn5Yel8ABVsK6T5sYxFcIgBfIgAzccYiAHI3MGYCBAyMyeZDaQA6b0YM6HRYb24VgG4Wlra+gdhh+lHxydwAChWAYjiwAkknwVuxug82kRmAYmFtwXlIZDmBUwC5RAhVotsMsAEbjADuBGS7WYnSIuGwPGMjD8YCihQpVJp-nkMRc-XwKRAikObg8RAiCxiAAlYNBoNgwYMUWiRABCGL0kTMAT+L4xADi2DwRHqfBEiWkMQAlKCIYi+dAMStAi4AIxhSJsjnJNKZe2pM0IxG9UQEbBwa3YaoiL2FREzdgQmbGu55TIlQRgHJ5ECRkAESpsBA2OwOJziHgEOScAG08Q4vAVbq6gzSRIYgCsAAZmLQiAAmWCMG6zNOcRNCawgWz2RzSZwgABW9DQ9EHAVLPnLSH7GazhHZsDAcgQSgH-CTu-3sWLgz4hG34kkKsphH9fGPylixkz2JI28g9l3wHTEA6fA7tY6aVNUP4gYQQE5iOebjpOoGEIeQ5Lv4K4SOmGCPhAojiHazZpkAA)

Please refer to the [SDK documentations](https://livecodes.io/docs/sdk) for details.

Comprehensive [documentations](https://livecodes.io/docs/) are available with live demos, code samples and screenshots.

LiveCodes is free with unlimited usage, no ads and no account required. It can be easily [self-hosted](https://livecodes.io/docs/features/self-hosting) (if you want), and can be used for commercial projects ([MIT license](https://livecodes.io/docs/license)).

Disclosure: obviously, I'm the author of LiveCodes.

I thought you would be interested. Otherwise, please feel free to close this.

**Comment by PierreQuentel** [comment]
Thanks Hatem, I have added a link to LiveCodes in the [Brython in the wild](https://github.com/brython-dev/brython/wiki/Brython-in-the-wild) section.
